CAIPs/caip-stream-sync.md

```diff
@@ -0,0 +1,193 @@
+
+## Specification
+<!--The technical specification should describe the standard in detail. The specification should be detailed enough to allow competing, interoperable implementations. -->
+
+### Protocol overview
+
+The stream sync protocol utilizes three main mechanisms in order to synchronize event streams between nodes. First of all the libp2p DHT is used to look up other nodes that provide the relevant streams, second a protocol for querying and gossiping tips ([CID](https://github.com/multiformats/cid) of latest event), and third a protocol for synchronizing events in verification order.
+
+### Protocol messages
+
+The stream sync protocol has four main message types that nodes send to each other. It’s implemented as a libp2p sub-protocol, and is inspired by [ipfs bitswap](https://docs.ipfs.tech/concepts/bitswap/).
+
+**Libp2p protocol id:**
+
+```
+/caip/streamsync/1.0.0
+```
+
+#### Synchronizing tips
+
+The following two protocol messages are used to query and gossip tips.
+
+**SubscribeSet**
+
+This message is used to subscribe to a set of streams. 
+
+```ipldsch
+type SubscribeSet struct {
+  separatorKey nullable String
+  separatorValue nullable Bytes
+  streamFilter nullable Bytes
+  controllerFilter nullable Bytes
+} representation tuple
+```
+
+*Properties:*
+
+* `separatorKey` - a string key that has to be set in the *genesis event header* of a matching stream
+* `separatorValue` - the value set for the `separatorKey`
+* `streamFilter` - a byte array which streamids in this set must end with, e.g. if this property contains 3 bytes all streamids that ends with these three bytes are part of the set
+* `controllerFilter` - A bloom filter populated with the DIDs that the set includes
+
+*Invariants:*
+
+* If `separatorKey` and `separatorValue` are null, `streamFilter` must be a full streamId
+* If `separatorKey` and `separatorValue` are set, `shardFilter` and `ctrlFilter` are optional
+
+**HaveTips**
+
+A response to a *SubscribeSet* message or an update sent to known subscribed peers containing the latest known tips of the given stream set.
+
+```ipldsch
+type HaveTips struct {
+  tips [Tips]
+  separatorKey nullable String
+  separatorValue nullable Bytes
+  streamFilter nullable Bytes
+  ctrlFilter nullable Bytes
+} representation tuple
+
+type Tips struct {
```

Comment:
StreamTips?

Reply:
trying to figure out a way for the struct to not have the same name as the field within the struct